lib/rest_4.js

```diff
@@ -408,19 +407,23 @@ function* compile(compileList, node) {
   return result;
 }
 
-function* isCompiled(contractName) {
+function* isCompiledByName(contractName) {
```

Comment:
I worry that isCompiledByName can be misleading, it could possibly return true after you've updated the contract source code. If this was intentional then everything looks good to me!